src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2165,10 +2164,7 @@ public override ITypeParameterSymbol GetDeclaredSymbol(TypeParameterSyntax typeP
                         return this.GetTypeParameterSymbol(typeSymbol.TypeParameters, typeParameter).GetPublicSymbol();
 
                     case MethodSymbol methodSymbol:
-                        return (this.GetTypeParameterSymbol(methodSymbol.TypeParameters, typeParameter) ??
-                            ((object)methodSymbol.PartialDefinitionPart == null
```

Comment:
I think this check didn't make sense for the same reason as https://github.com/dotnet/roslyn/pull/73603/files#r1617956185.